### PR TITLE
Implement MLJ inverse_transform

### PR DESCRIPTION
**Pull request by Kolaru**
This implement the `inverse_transform` interface for the decomposition method which have a `reconstruct` method.

In addition I used a bit of metaprograming to remove a bit of duplicated code in the interfaces.

### CODE REVIEW

File: src/models/decomposition_models.jl [hunk 1]
```diff
@@ -362,21 +318,47 @@ function MMI.fit(model::FactorAnalysis, verbosity::Int, X)
...
+    end
+
+    @eval function MMI.transform(::$M, fr::$MFitResultType, X)
+        # X is n x d, need to transpose twice
+        Xarray = MMI.matrix(X)
+        Xnew = transpose(MS.transform(fr, transpose(Xarray)))
+        return MMI.table(Xnew, prototype=X)
+    end
+
+    if hasmethod(MS.reconstruct, Tuple{MFitResultType{Float64}, Matrix{Float64}})
```
Comment (OkonSamuel) [inline src/models/decomposition_models.jl:353]:
```suggestion
    if hasmethod(MS.reconstruct, Tuple{MFitResultType{T}, AbstractMatrix{T}} where {T} )
```
not really a big deal but could you replace with above.

**Comment by OkonSamuel** [comment]
Thanks a lot @Kolaru. 
LGTM.
just that one thing above and this PR is ready to be merged.

**Comment by Kolaru** [comment]
Everything should be good now.

**Review by OkonSamuel — APPROVED** [review]

**Review by ablaom — APPROVED** [review]
@Kolaru Awesome contribution!

Thanks for your extreme patience. Somehow this PR fell off my radar.

**Pull request merged**